From 021b9fab2ba58dde338658052b45618fcbb8ebc7 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Wed, 1 Apr 2026 16:43:45 +0100
Subject: [PATCH] cow: Add name of the temporary file to debug output

If -D cow.verbose=1 display the name of the temporary file.  The file
is deleted so it does not appear in the filesystem.

(cherry picked from commit d415227d30ce7056ded227bb9490e937de230ae1)
---
 filters/cow/blk.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/filters/cow/blk.c b/filters/cow/blk.c
index 96d613e2..1012fb5e 100644
--- a/filters/cow/blk.c
+++ b/filters/cow/blk.c
@@ -198,6 +198,9 @@ create_temporary_file (void)
     return -1;
   }
 
+  if (cow_debug_verbose)
+    nbdkit_debug ("cow: created temporary file for overlay: %s", filename);
+
   unlink (filename);
   free (filename);
 
-- 
2.47.3

